src/components/Routes/Track/Applications/Applications.js

```diff
@@ -16,56 +26,132 @@ class Applications extends Component {
 
   count = 0;
   componentDidUpdate = (prevProps) => {
-    let lengths = true;
-    let notEqual = true;
+    let equal = false;
     if(this.count > 0) {
-      lengths = prevProps.savedApplications.length === 0 && this.props.savedApplications.length === 0;
-      notEqual = prevProps.savedApplications === this.props.savedApplications;
+      equal = isEqual(prevProps.savedApplications, this.props.savedApplications);
     }
 
-    if(this.props.isAuthenticated && lengths && notEqual) {
+    if(this.props.isAuthenticated && !equal) {
       this.props.getSavedApplications(this.props.token, this.props.userId);
+
+      if(!isEqual(this.props.savedApplications, this.state.savedApplications)) {
+        this.setState({
+          savedApplications: this.props.savedApplications
+        })
+      }
     }
 
     this.count++;
   }
 
+  dragStart = (event, index) => {
+    this.draggedItem = this.state.savedApplications[index];
+    event.dataTransfer.effectAllowed = "move";
+    event.dataTransfer.setData("text/html", event.target.parentNode);
+    event.dataTransfer.setDragImage(event.target.parentNode, 20, 20);
+  };
+
+  dragOver = (index) => {
+    const draggedOverItem = this.state.savedApplications[index];
+
+    if (this.draggedItem !== draggedOverItem) {
+      let savedApplications = this.state.savedApplications.filter(savedApplication => savedApplication !== this.draggedItem);
+
+      savedApplications.splice(index, 0, this.draggedItem);
+
+      this.setState({
+        savedApplications: savedApplications
+      });
+    }
+  };
+
+  save = () => {
+    const savedApplications = {};
+
+    this.state.savedApplications.forEach((savedApplication, index) => {
```

Comment:
@shoelessness Is the code faster this way? Just not sure why you're suggesting a reduce here.

Reply:
Calling `.reduce` means you can store the method’s result to a variable. This is a more clear pattern than initializing a variable and modifying it in a loop. I suggest this as a pattern in general. It’s also why I’ve suggested other methods instead of the for…in loops.

Reply:
@shoelessness ahh ok that makes sense. Thanks for the explanation!

Reply:
Yeah, it mostly has to do with the concept of boundaries within code.

Given:
- Scope A
  - Nested scope B

Anything that will be used elsewhere within A should only be modified within B in such a way that the purpose of the transform is clear to the rest of A. You shouldn't have to go digging in B unless you need to really change the guts of the transform. In this case, it would make it even more clear if the reducer callback (`(savedAppsResult, currentApp) => ({...` was abstracted to a named function and then referenced within this scope. Giving the name to the function gives clear meaning to the scope in A.

Reply:
Also, [more on reduce here](https://jrsinclair.com/articles/2019/functional-js-do-more-with-reduce/)